network/shuffle/src/main/java/org/apache/spark/network/shuffle/RetryingBlockFetcher.java

```diff
@@ -63,6 +64,9 @@
   /** Shared executor service used for waiting and retrying. */
   private static final ExecutorService executorService = Executors.newCachedThreadPool(
     NettyUtils.createThreadFactory("Block Fetch Retry"));
+  static {
```

Comment:
This change doesn't do anything, because you have a core pool size of 0 anyway. Could you back up and state the problem you are trying to solve in the JIRA? let's close this for now.